bootstore/src/schemes/v0/storage.rs

```diff
@@ -0,0 +1,105 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Storage for the v0 bootstore scheme
+//!
+//! We write two pieces of data to M.2 devices in production via
+//! [`omicron_common::Ledger`]:
+//!
+//!    1. [`super::Fsm::State`] for bootstore state itself
+//!    2. A network config blob required for pre-rack-unlock configuration
+//!
+
+use super::{Fsm, FsmConfig, State};
+use camino::Utf8PathBuf;
+use omicron_common::ledger::{Ledger, Ledgerable};
+use serde::{Deserialize, Serialize};
+use sled_hardware::Baseboard;
+use slog::{info, Logger};
+
+/// A persistent version of `Fsm::State`
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PersistentFsmState {
+    pub generation: u64,
+    pub state: State,
+}
+
+impl Ledgerable for PersistentFsmState {
+    fn is_newer_than(&self, other: &Self) -> bool {
```

Comment:
This name is misleading, since it's `>=` instead of `>`.

Reply:
Good point. I'll change the check, although it doesn't actually matter much for this usage of the code as do checks out of band from the ledger.